scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1559,6 +1560,20 @@
                (counters/dec! in-order-queue-counter)
                (body-fn))))))
 
+(defn prepare-match-trigger-chan
+  "Calculate the required interval for match-trigger-chan and start the chimes"
+  [match-trigger-chan pools]
+  (let [{:keys [per-pool-match-interval-millis global-min-match-interval-millis]} (config/matching-settings)
+        match-interval-millis (-> per-pool-match-interval-millis 
+                                  (/ (count pools)) 
+                                  int 
+                                  (max global-min-match-interval-millis))]
+    (when (= match-interval-millis global-min-match-interval-millis)
```

Comment:
I would remove the (max global-min-match-interval-millis). The target is having each pool visited at least as often as per-pool-match-interval-millis. The impact of match-interva-millis being too small (i.e., less than global-min-match-intetrval-millis) is increase of CPU (and the fallout of such). The cost of match-interval-millis * (count pools) too high is we have stale offers. The (when ...) then warns when match-interval-millis is less than **global-min-match-interval-millis-warn-threshold**. (I renamed it to indicate its purpose.)

(Also, related, if you don't agree with the logic above. We cannnot simutaneously have global-min-match-interval-millis and per-pool-match-interval-millis. We should call it per-pool-match-interval-millis-target.)

Reply:
I'm ok calling it per-pool-match-interval-millis-target but leaving the max

The max warns, but also stops CPU usage from going up if for some reason we didn't or couldn't act on the warning right away. Protects against adding a large number of pools at once.

Reply:
With the tradeoff of breaking mesos if too many pools get added at once. Which is more important: Saving CPU or not breaking mesos?

Reply:
If we don't save CPU we might break everything, including mesos. Mesos loses either way

Reply:
Fair. Only probably a concern if we get to 100+ pools and we're still using mesos.

But can you please rename it to per-pool-match-interval-millis-target  because its a target. We may miss it.